src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            // Unable to determine the correct vote with no metrics.
+            if (metrics == null || metrics.Length == 0)
+            {
+                return status;
+            }
+
+            // We shouldn't assign more workers than there are partitions
+            // This check is first, because it is independent of load or number of samples.
+            int partitionCount = metrics.Last().PartitionCount;
+            if (partitionCount > 0 && partitionCount < workerCount)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"WorkerCount ({workerCount}) > PartitionCount ({partitionCount}).");
+                _logger.LogInformation($"Number of instances ({workerCount}) is too high relative to number " +
+                                       $"of partitions for EventHub entity ({_eventHubName}, {partitionCount}).");
+                return status;
+            }
+
+            // At least 5 samples are required to make a scale decision for the rest of the checks.
+            if (metrics.Length < NumberOfSamplesToConsider)
+            {
+                return status;
+            }
+
+            // Maintain a minimum ratio of 1 worker per 1,000 unprocessed events.
+            long latestEventCount = metrics.Last().EventCount;
+            if (latestEventCount > workerCount * 1000)
+            {
+                status.Vote = ScaleVote.ScaleOut;
+                _logger.LogInformation($"EventCount ({latestEventCount}) > WorkerCount ({workerCount}) * 1,000.");
+                _logger.LogInformation($"Event count for EventHub entity ({_eventHubName}, {latestEventCount}) " +
+                                       $"is too high relative to the number of instances ({workerCount}).");
+                return status;
+            }
+
+            // Check to see if the EventHub has been empty for a while. Only if all metrics samples are empty do we scale down.
+            bool isIdle = metrics.All(m => m.EventCount == 0);
+            if (isIdle)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"'{_eventHubName}' is idle.");
+                return status;
+            }
+
+            // Samples are in chronological order. Check for a continuous increase in unprocessed event count.
+            // If detected, this results in an automatic scale out for the site container.
+            if (metrics[0].EventCount > 0)
+            {
+                bool eventCountIncreasing =
+                IsTrueForLastN(
+                    metrics,
+                    NumberOfSamplesToConsider,
+                    (prev, next) => prev.EventCount < next.EventCount) && metrics[0].EventCount > 0;

```

Comment:
I don't think we need this additional "metrics[0].EventCount > 0" since we're in an if block that already asserted this condition?